frontend/src/components/LoginForm/LoginForm.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { UserContext } from '../../UserContext.js';
+import './LoginForm.css'
+
+const LoginForm = () => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const { updateUser } = useContext(UserContext);
+
+  const navigate = useNavigate();
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+
+    try {
+      // Make the login API request
+      const response = await fetch(`http://localhost:3000/users/login`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ username, password }),
+        credentials: 'include'
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        const loggedInUser = data.user;
+
+        // Update the user context
+        updateUser(loggedInUser);
+
+        // Navigate to the home page after successful login
+        navigate('/');
+      } else {
+        // Handle the login failure case
+        alert('Login failed');
+      }
+    } catch (error) {
+      // Handle any network or API request errors
+      alert('Login failed: ' + error);
+    }
+  };
+
+  return (
+    <div className='login-form-container'>
+      <form className="login-form" onSubmit={handleLogin}>
+        <h2>Login</h2>
+        <div className="form-group">
+          <label htmlFor="username">Username:</label>
+          <input
+            type="text"
+            id="username"
+            value={username}
+            onChange={(e) => setUsername(e.target.value)}
+            required
+          />
+        </div>
+        <div className="form-group">
+          <label htmlFor="password">Password:</label>
+          <input
+            type="password"
+            id="password"
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
+            required
+          />
+        </div>
+        <button type="submit">Login</button>
```

Comment:
What would happen if the user left `username` and `password` blank? Can we make this experience better, e.g disable if empty?

Reply:
![Screenshot 2024-07-01 at 9 32 08 AM](https://github.com/CristobalMe/Event_Finder/assets/112215849/b282432c-4a83-404c-ab2f-4c906a5c40c2)

Reply:
Done